service/s3/internal/customizations/presign_test.go

```diff
@@ -41,6 +41,7 @@ func TestPutObject_PresignURL(t *testing.T) {
 			},
 			expectMethod: "PUT",
 			expectSignedHeader: http.Header{
+				"Amz-Sdk-Request": []string{"attempt=1; max=1"},
```

Comment:
Why?

Reply:
it will be helpful to see the stack at these stages

pre-SRA auth call to `PresignPutObject`
```
	Finalize stack step
		PresignHTTPRequest
```

post-SRA auth call to `PresignPutObject`
```
	Finalize stack step
		ResolveAuthScheme
		GetIdentity
		ResolveEndpointV2
		ComputePayloadHash
		disableHTTPS
		AWSChecksum:ComputeInputPayloadChecksum
		setLegacyContextSigningOptions
		presignContextPolyfill
		PresignHTTPRequest
		Retry
		RetryMetricsHeader
```

post-this PR  call to `PresignPutObject`
```
	Finalize stack step
		ResolveAuthScheme
		GetIdentity
		ResolveEndpointV2
		ComputePayloadHash
		disableHTTPS
		AWSChecksum:ComputeInputPayloadChecksum
		Retry
		addInputChecksumTrailer
		RetryMetricsHeader
		setLegacyContextSigningOptions
		presignContextPolyfill
		PresignHTTPRequest
```


so, the retrymetrics middleware pre-SRA auth wasnt even in finalize (nor any of the other phases) in presign putobject calls. post-SRA the retry metrics middleware was added AFTER presigning, so with regards to this test, it didnt matter. now with this PR, the presigning middleware is the LAST middleware to run in finalize (which is what it was pre-SRA). so yes, we are reverting the middleware to its original state with respect to presigning being last. but retry metrics have additionally been added into the finalize phase (when no other middlewares were there previously)

Reply:
That all makes sense but it's still a behavioral change that we shouldn't be taking without justification, it doesn't have anything to do with the thing we're fixing here.